src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1625,39 +1621,14 @@ public sealed override async Task<HttpResponseMessage> SendAsync(HttpRequestMess
             }
             catch (Exception e)
             {
-                Exception replacementException = null;
-
                 if (e is IOException ||
                     e is ObjectDisposedException ||
                     e is Http2ProtocolException ||
                     e is InvalidOperationException)
                 {
-                    replacementException = new HttpRequestException(SR.net_http_client_execution_error, _abortException ?? e);
-                }
-                else if (e is OperationCanceledException oce)
-                {
-                    // If the operation has been canceled after the stream was allocated an ID, send a RST_STREAM.
-                    if (http2Stream != null && http2Stream.StreamId != 0)
-                    {
-                        http2Stream.Cancel();
-                    }
-
-                    if (cancellationToken.IsCancellationRequested && oce.CancellationToken != cancellationToken)
-                    {
-                        replacementException = new OperationCanceledException(oce.Message, oce, cancellationToken);
-                    }
+                    throw new HttpRequestException(SR.net_http_client_execution_error, e);
```

Comment:
Why is it ok to remove the `_abortException ??` here?

Reply:
Because the exception that propagates from failure above should be the correct exception to give to the user.

It's possible that there's some weird timing issues where (a) an exception happens above, and starts to propagate; (b) OnReset is called, and sets _resetException; (c) this causes the original exception to get overwritten. If the code above is failing for the same reason OnReset is being called -- i.e. connection failure or whatever -- then the same exception should get propagated both ways.